### ISSUE TITLE
build error in CSMIMcast.h

### ISSUE THREAD

**Issue by adambertsch** [IBM/CAST#23]
I was having trouble building on my branch, so I decided to build off of master first.  It looks like there's a type mismatch in the current master.  I don't know enough about the code base yet to say for certain this is the problem since I haven't chased the call path all the way yet (which is unfortunately really hard in the API due to the serialization stuff).

```
/g/g0/bertsch2/src/build/CAST/csmd/src/daemon/src/csmi_request_handler/csmi_mcast/CSMIMcast.h:94:34: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
             for( uint32_t i=0; i < _Data->num_nodes; ++i)
```

Not clear if it is intentional or not, but in different structures num_nodes is sometimes defined signed, and sometimes unsigned.  csmi_allocation_step_t is signed, while the others are all unsigned.

```
[bertsch2@butte5 CAST]$ grep -r 'int32_t num_nodes' *
csmd/src/daemon/src/csmi_request_handler/CSMIAllocationStepQueryDetails.cc:            int32_t num_nodes = output->steps[i]->num_nodes;
csmd/src/daemon/src/csmi_request_handler/CSMIJSRUNCMD.cc:        uint32_t num_nodes =  strtol(fields->data[1], nullptr, 10);
csmd/src/daemon/src/csmi_request_handler/csmi_mcast/CSMIMcastBB.h:    uint32_t num_nodes;
csmd/src/daemon/src/csmi_request_handler/csmi_mcast/CSMIMcastJSRUN.h:    uint32_t num_nodes;
csmi/include/csmi_type_wm.h:    uint32_t num_nodes; /**< Number of nodes, size of @ref compute_nodes. */
csmi/include/csmi_type_wm.h:    uint32_t num_nodes; /**< Number of nodes in the step, size of @ref compute_nodes. */
csmi/include/csmi_type_wm.h:    uint32_t num_nodes; /**< Number of nodes in allocation, size of @ref node_accounting. */
csmi/include/csmi_type_wm.h:    int32_t num_nodes; /**< Number of nodes, size of @ref compute_nodes.*/
csmi/src/wm/include/csmi_wm_type_internal.h:    uint32_t num_nodes; /**< Number of nodes, size of @ref compute_nodes. */
csmi/src/wm/include/csmi_wm_type_internal.h:    int32_t num_nodes; /**< Number of nodes, size of @ref compute_nodes.*/
[bertsch2@butte5 CAST]$ 
```

**Comment by mew2057** [comment]
I have a pull request fixing this issue, but it does change the struct header.

**Comment by mew2057** [comment]
This was fixed in 1.2.0 I believe.
